pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -24,130 +24,185 @@ import (
 	"google.golang.org/grpc"
 	grpccodes "google.golang.org/grpc/codes"
 	grpcstatus "google.golang.org/grpc/status"
-	"k8s.io/klog/v2"
 
+	"k8s.io/klog/v2"
 	drapbv1alpha2 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
 	drapb "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
 )
 
 const PluginClientTimeout = 45 * time.Second
 
-// draPluginClient encapsulates all dra plugin methods.
-type draPluginClient struct {
-	pluginName string
-	plugin     *Plugin
+type (
+	nodeResourceManager interface {
+		Prepare(context.Context, *grpc.ClientConn, *plugin, *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error)
+		Unprepare(context.Context, *grpc.ClientConn, *plugin, *drapb.NodeUnprepareResourcesRequest) (*drapb.NodeUnprepareResourcesResponse, error)
+	}
+
+	v1alpha2NodeResourceManager struct{}
+	v1alpha3NodeResourceManager struct{}
+)
+
+var nodeResourceManagers = map[string]nodeResourceManager{
+	v1alpha2Version: v1alpha2NodeResourceManager{},
+	v1alpha3Version: v1alpha3NodeResourceManager{},
+}
+
+func (v1alpha2rm v1alpha2NodeResourceManager) Prepare(ctx context.Context, conn *grpc.ClientConn, _ *plugin, req *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error) {
+	nodeClient := drapbv1alpha2.NewNodeClient(conn)
+	response := &drapb.NodePrepareResourcesResponse{
+		Claims: make(map[string]*drapb.NodePrepareResourceResponse),
+	}
+
+	for _, claim := range req.Claims {
+		res, err := nodeClient.NodePrepareResource(ctx,
+			&drapbv1alpha2.NodePrepareResourceRequest{
+				Namespace:      claim.Namespace,
+				ClaimUid:       claim.Uid,
+				ClaimName:      claim.Name,
+				ResourceHandle: claim.ResourceHandle,
+			})
+		result := &drapb.NodePrepareResourceResponse{}
+		if err != nil {
+			result.Error = err.Error()
+		} else {
+			result.CDIDevices = res.CdiDevices
+		}
+		response.Claims[claim.Uid] = result
+	}
+
+	return response, nil
+}
+
+func (v1alpha2rm v1alpha2NodeResourceManager) Unprepare(ctx context.Context, conn *grpc.ClientConn, _ *plugin, req *drapb.NodeUnprepareResourcesRequest) (*drapb.NodeUnprepareResourcesResponse, error) {
+	nodeClient := drapbv1alpha2.NewNodeClient(conn)
+	response := &drapb.NodeUnprepareResourcesResponse{
+		Claims: make(map[string]*drapb.NodeUnprepareResourceResponse),
+	}
+
+	for _, claim := range req.Claims {
+		result := &drapb.NodeUnprepareResourceResponse{}
+		if _, err := nodeClient.NodeUnprepareResource(ctx,
+			&drapbv1alpha2.NodeUnprepareResourceRequest{
+				Namespace:      claim.Namespace,
+				ClaimUid:       claim.Uid,
+				ClaimName:      claim.Name,
+				ResourceHandle: claim.ResourceHandle,
+			}); err != nil {
+			result.Error = err.Error()
+		}
+		response.Claims[claim.Uid] = result
```

Comment:
Can we split this to by symmetrical with what you have for `Prepare`, i.e.

```suggestion
		_, err := nodeClient.NodeUnprepareResource(ctx,
			&drapbv1alpha2.NodeUnprepareResourceRequest{
				Namespace:      claim.Namespace,
				ClaimUid:       claim.Uid,
				ClaimName:      claim.Name,
				ResourceHandle: claim.ResourceHandle,
			})
		result := &drapb.NodeUnprepareResourceResponse{}
		if err != nil {
			result.Error = err.Error()
		}
		response.Claims[claim.Uid] = result
```

Reply:
done in b55b0297bad